R/FilterStateDate.R

```diff
@@ -92,35 +97,71 @@ DateFilterState <- R6::R6Class( # nolint
     #' Initialize a `FilterState` object
+    #'
     #' @param x (`Date`)\cr
     #'   values of the variable used in filter
     #' @param x_reactive (`reactive`)\cr
-    #'   a `reactive` returning a filtered vector or returning `NULL`. Is used to update
-    #'   counts following the change in values of the filtered dataset. If the `reactive`
-    #'   is `NULL` counts based on filtered dataset are not shown.
-    #' @param varname (`character`, `name`)\cr
-    #'   name of the variable
-    #' @param varlabel (`character(1)`)\cr
-    #'   label of the variable (optional).
+    #'   returning vector of the same type as `x`. Is used to update
+    #'   counts following the change in values of the filtered dataset.
+    #'   If it is set to `reactive(NULL)` then counts based on filtered
+    #'   dataset are not shown.
     #' @param dataname (`character(1)`)\cr
-    #'   optional name of dataset where `x` is taken from
+    #'   optional name of dataset where `x` is taken from. Must be specified
+    #'   if `extract_type` argument is not empty.
+    #' @param varname (`character(1)`)\cr
+    #'   name of the variable.
+    #' @param choices (`atomic`, `NULL`)\cr
+    #'   vector specifying allowed selection values
+    #' @param selected (`atomic`, `NULL`)\cr
+    #'   vector specifying selection
+    #' @param keep_na (`logical(1)`, `NULL`)\cr
+    #'   flag specifying whether to keep missing values
+    #' @param keep_inf (`logical(1)`, `NULL`)\cr
+    #'   flag specifying whether to keep infinite values
+    #' @param fixed (`logical(1)`)\cr
+    #'   flag specifying whether the `FilterState` is initiated fixed
+    #' @param extras (`named list` or `NULL`) of `character` vectors\cr
+    #'   storing additional information on this filter state
+    #' @param varlabel (`character(0)`, `character(1)`)\cr
+    #'   label of the variable (optional)
     #' @param extract_type (`character(0)`, `character(1)`)\cr
     #' whether condition calls should be prefixed by dataname. Possible values:
     #' \itemize{
     #' \item{`character(0)` (default)}{ `varname` in the condition call will not be prefixed}
     #' \item{`"list"`}{ `varname` in the condition call will be returned as `<dataname>$<varname>`}
     #' \item{`"matrix"`}{ `varname` in the condition call will be returned as `<dataname>[, <varname>]`}
     #' }
+    #'
     initialize = function(x,
                           x_reactive = reactive(NULL),
+                          dataname,
                           varname,
+                          choices = NULL,
+                          selected = NULL,
+                          keep_na = NULL,
+                          keep_inf = NULL,
+                          fixed = FALSE,
+                          extras = NULL,
+                          dataname_prefixed = character(0),
                           varlabel = character(0),
-                          dataname = NULL,
                           extract_type = character(0)) {
```

Comment:
This constructor is getting too complex in my opinion. I still think `dataname_prefixed` and `extract_type`
 can be removed. This would mean that onlu `x`, `x_reactive` and `varlabel` would be only fields not covered in the API.
 
I would change `get_call`, `get_varname_prefixed` to have `dataname` and `extract_type` and change `add_keep_na_call(filter_call)` to `get_keep_na_call(dataname, extract_type)` (and the same with `add_keep_inf_call`).
 Thanks to this `get_keep_na_call` would have the same arguments as `get_call` and would be easier to understand.
 
```suggestion
                          varlabel = character(0)) {
```